src/Analysis/Engine/Impl/DependencyResolution/PathResolverSnapshot.cs

```diff
@@ -0,0 +1,280 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.PythonTools.Analysis.Infrastructure;
+using Microsoft.PythonTools.Parsing;
+
+namespace Microsoft.PythonTools.Analysis.DependencyResolution {
+    internal partial struct PathResolverSnapshot {
+        private static readonly bool IgnoreCase = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        private readonly PythonLanguageVersion _pythonLanguageVersion;
+        private readonly Node[] _roots;
+        public int Version { get; }
+
+        public PathResolverSnapshot(PythonLanguageVersion pythonLanguageVersion) {
+            _pythonLanguageVersion = pythonLanguageVersion;
+            _roots = Array.Empty<Node>();
+            Version = 0;
+        }
+
+        private PathResolverSnapshot(PythonLanguageVersion pythonLanguageVersion, Node[] roots, int version) {
+            _pythonLanguageVersion = pythonLanguageVersion;
+            _roots = roots;
+            Version = version;
+        }
+
+        public IAvailableImports GetAvailableImportsFromAbsolutePath(in string modulePath, in IEnumerable<string> path) {
+            if (!TryFindModule(modulePath, out var lastArc, out _, out _)) {
+                return default;
```

Comment:
Does it return null? Perhaps make static empty instance of AvaiableImports returning empty objects?

Reply:
Not sure about that, empty instance will require some default values. Need to get to the next step when it will be used by walker.

Reply:
OK